### PR TITLE
chore: update 07-tendermint GetConsensusState to return bool over error

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Updates the `07-tendermint` helper `GetConsensusState` to return `(*ConsensusState, bool)` in favour of `(*ConsensusState, error)`

closes: #1140 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by damiannolan** [comment]
Happy to leave this on the back burner until the higher priority PRs are merged. Should minimise disruptions with merge conflicts..etc

File: modules/light-clients/07-tendermint/types/store.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/store.go:69]: should we panic?

File: modules/light-clients/07-tendermint/types/store.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/store.go:74]: ditto?

File: modules/light-clients/07-tendermint/types/store.go
Comment (damiannolan) [reply modules/light-clients/07-tendermint/types/store.go:69]: Wondered that too myself, was considering putting it in this PR, but I figured I'd wait to discuss here. 

I think its reasonable enough to panic if unmarshalling or type assertion fails. It *should* be unreachable code anyway, but in the event it fails to unmarshal for example something has clearly gone wrong..

Happy to panic here if you are. Any thoughts @seantking ?

Comment (seantking) [reply modules/light-clients/07-tendermint/types/store.go:69]: I think panic here makes sense. I don't see a reason to err gracefully in this situation :thinking:

**Review by seantking — APPROVED** [review]
LGTM once @colin-axner suggestion is applied.

File: modules/light-clients/07-tendermint/types/store.go
Comment (houshengbo) [inline modules/light-clients/07-tendermint/types/store.go:69]: When this functions returns an error, at least something descriptive was returned, and that error is not fatal to use panic.
Could you elaborate more on why panic is being used here. Is the error here fatal to crash the system?
If malformed bytes is unlikely to happen, do we have validation on cdc before calling UnmarshalConsensusState? I think this validation can make sure we do not run into this fatal error.

Comment (colin-axner) [reply modules/light-clients/07-tendermint/types/store.go]: The SDK gracefully handles panic during transaction execution. The error returned for the transaction usually wipes the original error message but indicates that something very unexpected occurred. In this case, for the panic to occur, there must be a bug in the existing code (we must of set malformed bytes in state)

File: modules/light-clients/07-tendermint/types/misbehaviour_handle.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/misbehaviour_handle.go:85]: comment can be removed, below as well

File: modules/light-clients/07-tendermint/types/misbehaviour_handle.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/misbehaviour_handle.go:88]:
```suggestion
		return sdkerrors.Wrapf(clienttypes.ErrConsensusStateNotFound, "could not get trusted consensus state from clientStore for Header1 at TrustedHeight: %s", misbehaviour.Header1.TrustedHeight)
```

File: modules/light-clients/07-tendermint/types/misbehaviour_handle.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/misbehaviour_handle.go:94]:
```suggestion
		return sdkerrors.Wrapf(clienttypes.ErrConsensusStateNotFound, "could not get trusted consensus state from clientStore for Header2 at TrustedHeight: %s", misbehaviour.Header2.TrustedHeight)
```

File: modules/light-clients/07-tendermint/types/store.go
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/store.go:60]: godoc needs to be updated. Should we make this function private?

File: modules/light-clients/07-tendermint/types/store_test.go
Comment (colin-axner) [reply modules/light-clients/07-tendermint/types/store_test.go:68]: lgtm

File: modules/light-clients/07-tendermint/types/store.go
Comment (damiannolan) [reply modules/light-clients/07-tendermint/types/store.go]: The panic will essentially bubble up the callstack of the `UpdateClient` handler and the [grpc recovery interceptor](https://github.com/cosmos/cosmos-sdk/blob/master/baseapp/grpcserver.go#L81) in the cosmos sdk will catch and recover 👍

File: modules/light-clients/07-tendermint/types/store.go
Comment (damiannolan) [reply modules/light-clients/07-tendermint/types/store.go:60]: Good spot

Comment (damiannolan) [reply modules/light-clients/07-tendermint/types/store.go:60]: Making private makes a bit more awkward for testing. We would actually need to remove the test function for this.
`getConsensusState` will still get coverage from a number of other tests but testing the panic scenarios then isn't really possible.

What do you think? I don't really have any strong feelings about making it public/private

Comment (colin-axner) [reply modules/light-clients/07-tendermint/types/store.go]: lets leave for now

**Review by colin-axner — APPROVED** [review]
Wahoo, nice!

**Pull request merged**
